genienlp/train.py

```diff
@@ -516,16 +516,25 @@ def train(
                 round_progress = f'round_{rnd}:' if rounds else ''
 
                 # param update
-                loss, grad_norm = train_step(
-                    model,
-                    batch,
-                    iteration,
-                    opt,
-                    devices,
-                    lr_scheduler=lr_scheduler,
-                    grad_clip=args.grad_clip,
-                    gradient_accumulation_steps=args.gradient_accumulation_steps,
-                )
+                try:
+                    loss, grad_norm = train_step(
+                        model,
+                        batch,
+                        iteration,
+                        opt,
+                        devices,
+                        lr_scheduler=lr_scheduler,
+                        grad_clip=args.grad_clip,
+                        gradient_accumulation_steps=args.gradient_accumulation_steps,
+                    )
+                except RuntimeError as e:
```

Comment:
But this will give poor results during training, as only part of the training data is seen. Validation results might also be different on different hardware, without users noticing.

Reply:
> But this has the potential to give poor results during training, as only part of the training data is seen.

Not necessarily. I have run experiments before and the results are good. Usually, only a few examples are problematic and will be skipped.

> Validation results might also be different on different hardware, without users noticing.

Yeah, this is why I think making it optional is reasonable with a cmd-line argument that is off by default.

> I think it is reasonable to assume that if people cannot fit even a single example of a dataset on their GPU, they should find another way to run that code. Do we have an issue like that where even single examples don't fit?

Yeah if you're trying very big models there's no point in using this option. The option is useful when you get very few OOMs e.g. once or twice per epoch.